src/gql/queries/patch.ts

```diff
@@ -1,27 +1,5 @@
 import gql from "graphql-tag";
-
-export interface Patch {
```

Comment:
moved these types to `types/patch.ts`

Reply:
We should leave the Patch interface in this file because the structure is specific to the fields in the gql string literal. As our codebase scales, it is possible that we will have another query that relies on Patch with different required fields so it could become cumbersome if we include all those interfaces together in `types/patch.ts`

Reply:
yes was thinking this too

Reply:
I don't necessarily agree, i think it makes more sense to keep our commonly reused types located in one area so we have a single source of truth for what the type for a certain query or field is. Its more error prone if we start recreating types on a per use basis since its easy to mistype a field.  
It shouldn't be an issue if we have different queries that rely on different subsets of fields with in a specific type since the type checking will still be valid on the fields that we do request in our queries.

Reply:
what arjun is saying is that this type is specifically the type returned by the graphql query and therefore belongs in the graphql namespace.

Reply:
all of the types for gql are currently in the gql folder. therefore it would make less sense organizationally to put one gql type in a different place than the rest of them. consistency is key when it comes to what does and does not make sense in a project's file structure. therefore putting this one gql type in a different place than the rest of them goes against your own point of a single source of truth. the point you provide goes against your own argument because doing it the way you suggest creates two sources of truth for gql types.

Reply:
The type exported from the graphql query is still defined in this file. 
https://github.com/evergreen-ci/spruce/blob/3956831786a197bf1bebbbd5812f9742e6d95fd5/src/gql/queries/patch.ts#L4

I just moved the type used by it

Reply:
The reason i moved the patch type was because the commitQueue returns a patch  type and  i thought it made more sense if they both implemented the same patch type.

Reply:
it's still a graphql type bc it represents the shape of a query. i see your point tho and do think there could be better file structure around types. however not certain that moving this one gql type to a different location is way to go. 

i would like to code generate the types from the gql schema soon anyways. and in which case all gql types will be in a single file in the graphql directory.

Reply:
we talked offline and decided that best way forward is to codegen the TS types for gql. jira ticket: https://jira.mongodb.org/browse/EVG-7866